src/WordPress/JsonMapper/ArrayInformation.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace WordPress\JsonMapper;
+
+class ArrayInformation implements \JsonSerializable
+{
+	/** @var bool */
+	private $isArray;
+	/** @var int */
+	private $dimensions;
+
+	private function __construct(bool $isArray, int $dimensions)
+	{
+		$this->isArray = $isArray;
+		$this->dimensions = $dimensions;
+	}
+
+	public static function notAnArray(): self
+	{
+		return new self(false, 0);
+	}
+
+	public static function singleDimension(): self
+	{
+		return new self(true, 1);
+	}
+
+	public static function multiDimension(int $dimension): self
+	{
+		return new self(true, $dimension);
+	}
+
+	public function isArray(): bool
+	{
+		return $this->isArray;
+	}
+
+	public function getDimensions(): int
+	{
+		return $this->dimensions;
+	}
+
+	public function isMultiDimensionalArray(): bool
```

Comment:
This method seems to be unused, let's remove it.

Reply:
Sure. Was made obsolete by simplifying the recursive methods in the Property Mapper.